### PR TITLE
TINY-4770: Clamped child node selection, added tests.

### LINKED ISSUE #5488: Clear highlights when the selection doesn't have highlighted text cause an error
When I use the backcolor plugin, I highlight the text, when I try to clear the highlight and I grab text that is not highlighted, and error appears on console and the highlight is not clear. This problem usually happens when you select the last part of the highlighted text with some text that is not part of the highlight.

This is how the text is selected when I want to clear the highlight.
Screen Shot 2020-03-09 at 11 41 25

This is the error I get on Chrome
Screen Shot 2020-03-09 at 11 41 42

This is the error I get on Firefox
Screen Shot 2020-03-09 at 11 42 04

I'm using tiny 5.2 version. I try this on Firefox 72.0.2 (64-bit) & Chrome 80.0.3987.132 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by HAFRMO**
Fixes #5488

### CODE REVIEW

File: modules/tinymce/src/core/test/ts/browser/fmt/RemoveHighlightFormatTest.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+interface Editor {
+  formatter: {
+    remove: (target: string, any: any) => void;
+  };
+}
+UnitTest.asynctest('browser.tinymce.core.fmt.RemoveHighlightFormatTest', (success, failure) => {
+  SilverTheme();
+
+  const sRemoveHighlight = function (editor: Editor) {
+    return Step.sync(function () {
```
[the remark this answers is not in the document]
Comment (lnewson) [reply modules/tinymce/src/core/test/ts/browser/fmt/RemoveHighlightFormatTest.ts]: Sorry, this is removing not toggling, in that case you'd use the following instead:
```
tinyApis.sExecCommand('RemoveFormat')
```

[the remark this answers is not in the document]
Comment (spocke) [reply modules/tinymce/src/core/test/ts/browser/fmt/RemoveHighlightFormatTest.ts]: Double check though that removeformat does indeed fail as well without the fix.

File: modules/tinymce/src/core/test/ts/browser/fmt/RemoveHighlightFormatTest.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          sRemoveHighlight(editor),
+          tinyApis.sAssertContent('<p>Part 1 Part 2 Part 3 Part 4</p>'),
+          tinyApis.sAssertSelection([0, 0], 2, [0, 2], 4)
+        ])),
+      ])),
+    ], onSuccess, onFailure);
+  }, {
+    base_url: '/project/tinymce/js/tinymce',
+    plugins: '',
+    toolbar: '',
```
[the remark this answers is not in the document]
Comment (HAFRMO) [reply modules/tinymce/src/core/test/ts/browser/fmt/RemoveHighlightFormatTest.ts]: Understood, I'll remove it.

File: modules/tinymce/src/core/main/ts/selection/RangeWalk.ts [hunk 3]
```diff
@@ -26,6 +26,18 @@ const getEndChild = function (container: Node, index: number) {
   return childNodes[index] || container;
 };
 
+const clampToExistingChildren = function (container: Node, index: number) {
```
[the remark this answers is not in the document]
Comment (HAFRMO) [reply modules/tinymce/src/core/main/ts/selection/RangeWalk.ts]: That could indeed work. I'll do that. I wasn't quite sure how we prefer to handle this kind of code duplication.

File: modules/tinymce/src/core/test/ts/browser/fmt/RemoveHighlightFormatTest.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (HAFRMO) [reply modules/tinymce/src/core/test/ts/browser/fmt/RemoveHighlightFormatTest.ts]: Will do. Checked that it fails without the fix ( it does ).

**Pull request merged**